app/controllers/lesson_contents_controller.rb

```
@@ -0,0 +1,42 @@
+require 'zip'
+
+class LessonContentsController < ApplicationController
+  # FIXME: Get rid of this, something weird is going on right now
+  skip_before_action :authenticate_user!
+  skip_before_action :ensure_admin!
+  skip_before_action :verify_authenticity_token
+
+  def new
+    @lesson_content = LessonContent.new
+  end
+
+  def create
+    # TODO: Should we make state optional, e.g. allow creating lessons contents
+    # outside of Canvas's LTI link extension?
+    # https://app.asana.com/0/1174274412967132/1184800386160070
+    params.require([:state, :lesson_content_zipfile])
+    lti_launch = LtiLaunch.current(params[:state])
+
+    @lesson_content = LessonContent.new(lesson_content_create_params)
+    @lesson_content.save
```

We should get in the habit of calling `save!` which if it fails will throw an exception. This will fail silently.